src/interpreter/memory.rs

```diff
@@ -85,9 +87,8 @@ impl MemoryInstance {
 	pub fn grow(&self, pages: u32) -> Result<u32, Error> {
 		let mut buffer = self.buffer.write();
 		let old_size = buffer.len() as u32;
-		match pages.checked_mul(LINEAR_MEMORY_PAGE_SIZE).and_then(|bytes| old_size.checked_add(bytes)) {
+		match calculate_memory_size(old_size, pages, self.maximum_size) {
 			None => Ok(u32::MAX),
-			Some(new_size) if new_size > self.maximum_size => Ok(u32::MAX),
 			Some(new_size) => {
 				buffer.extend(vec![0; (new_size - old_size) as usize]);
```

Comment:
this is going to be double allocation for grown size
better to force buffer to allocate once and then fill it with zeros without allocating another zero-buffer
or leave another todo)